baremetal/metal3machine_manager.go

```diff
@@ -169,6 +177,18 @@ func (m *MachineManager) IsProvisioned() bool {
 	return false
 }
 
+// IsBaremetalHostProvisioned returns true of if the provisioning state of the underlying baremetalhost is `Provisioned`.
+func (m *MachineManager) IsBaremetalHostProvisioned(ctx context.Context) bool {
+	host, _, err := m.getHost(ctx)
+	if err != nil {
+		return false
```

Comment:
shouldn't we return the error also?

Reply:
It makes for a nicer API on the caller side.  If we can't find a host, I think we can be pretty confident that the host isn't provisioned?

Reply:
As long as we're handling this error somewhere else (which we probably do), I guess it's fine?